src/views/open-api-docs-view-v2/components/content-with-permalink/index.tsx

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Link from 'next/link'
+import { IconLink16 } from '@hashicorp/flight-icons/svg-react/link-16'
+// Types
+import type { PropsWithChildren } from 'react'
+// Styles
+import s from './content-with-permalink.module.css'
+import classNames from 'classnames'
+
+/**
+ * Renders the provided `children` alongside a permalink `<a />` element.
+ *
+ * `ariaLabel` is required to ensure the permalink element has meaningful
+ * text for screen readers. For simple use cases where `children` is a
+ * string, the `ariaLabel` can be passed the same value as `children`.
+ * For more complex use cases, the consumer should determine what text
+ * would be appropriate to pass to `ariaLabel`.
+ *
+ * Note that this component does _not_ handle placing the provided `id`
+ * in the DOM. It requires the consumer to place the `id` on an appropriate
+ * element, typically an element rendered in the provided `children`.
+ */
+export function ContentWithPermalink({
```

Comment:
All of the new files in this PR are existing components copied over from the previous view. As mentioned in previous PRs, rationale is that once we've migrated to this new view, we can delete the old view folder. Copying the necessary components into the new view, rather than importing them from the old folder or trying to build some abstraction, is intended to make it easier to clean up the old view folder once we've migrated away from it.